src/utils/utils_common.h

```diff
@@ -96,6 +96,17 @@ static inline size_t align_size(size_t size, size_t alignment) {
     }
 }
 
+static inline void *align_ptr(void *ptr, size_t alignment) {
+    // align pointer to 'alignment' bytes
+    uintptr_t uptr = (uintptr_t)ptr;
+    uintptr_t rest = uptr & (alignment - 1);
+    if (rest) {
```

Comment:
I think you can avoid this check, you could probably copy the implementation from pmdk: https://github.com/pmem/pmdk/blob/e8d67f4b0d8c47d4e1d36cfcbd71726638b1f1e1/src/core/util.h#L73

Reply:
Done